src/main/java/com/google/devtools/build/lib/packages/TargetUtils.java

```diff
@@ -59,6 +62,17 @@ public boolean apply(String tag) {
     }
   };
 
+  // Not all the target level tags are propagated as execution requirements to the actions, because
+  // we do not want to propagate custom user's tags or create potential conflicts
+  // with execution requirements declared on the rules.
+  // See https://github.com/bazelbuild/bazel/issues/7766 for details.
+  private static final Predicate<String> TAGS_PROPAGATED_TO_EXEC_INFO =
+      tag -> tag.equals("no-remote")
```

Comment:
As somebody completely ignorant of these things my first reaction is hmm where can I find out what these do. Wdyt about making these constants, and adding javadoc and document (or reference existing documentation) from these constants?